src/util/misc.py

```diff
@@ -510,47 +510,51 @@ def visit_UnaryOp(self, node):
                 return self._replace_node("_upg_UnaryOp_Not", node) if isinstance(node.op, ast.Not) else node
 
         replacer = RewriteName()
-        root = ast.parse(expr.strip(), mode="eval").body
+        root = ast.parse(expr.strip(), mode="exec")
         visited = replacer.visit(root)
         return (ast_unparse(visited).strip(), SelfPrintEvalContext(replacer.replaces))
 
 
+GLOB_LIT = object()
```

Comment:
What about scoping it by being an attribute on the `literal_replace`
Also rename it to something more explicit.

```suggestion
literal_replace.WILDCARD = object()
```

Reply:
Done. 
Subtleties:
1. `ast.Attributes` needs a string to unparse correctly
    ```py
    >>> ast.unparse(ast.Attribute(ast.Name("y"), object()))
    Traceback (most recent call last):
      File "<input>", line 1, in <module>
        ast.unparse(ast.Attribute(ast.Name("y"), object()))
      File "/usr/lib/python3.12/ast.py", line 1791, in unparse
        return unparser.visit(ast_obj)
               ^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.12/ast.py", line 860, in visit
        return "".join(self._source)
               ^^^^^^^^^^^^^^^^^^^^^
    TypeError: sequence item 2: expected str instance, object found
    ```
2. Cannot really use `str is str`:
    ```py
    >>> x = "*"
    >>> y = "*"
    >>> x is y
    True
    ```
    ```py
    >>> class S(str):
    ...     pass
    ...
    >>> z = S("*")
    >>> z is x
    False
    ```
3. `ctx` is missing in manually created nodes:
    ```py
    >>> ast.parse("y.z", mode="eval").body.ctx
    <ast.Load object at 0x7c1cebe20bd0>
    >>> ast.Attribute(ast.Name("y"), "z").ctx
    Traceback (most recent call last):
      File "<input>", line 1, in <module>
        ast.Attribute(ast.Name("y"), "z").ctx
    AttributeError: 'Attribute' object has no attribute 'ctx'
    ```

Thus for 3/ and the sake of less verbosity we assume missing ctx means sort of a wildcard context.

Reply:
Using `_Symbol` works when creating a `ast.Attribute` record.

Does it also for other `_match_X` methods where it is used?

I would like to see them also be covered by a test.

Reply:
I will add tests for them.

Reply:
Done. Note that this just a nice to have: good for debugging if we need to print the pattern.